### PR TITLE
Update padding functions

### PR DESCRIPTION
**Pull request by dylanjw**
### What was wrong?

https://github.com/ethereum/eth-utils/pull/58 removed padding functions.

### How was it fixed?

Added wrappers around rjust, ljust methods.

#### Cute Animal Picture

Cute animal picture

### CODE REVIEW

File: eth_abi/utils/padding.py [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
+def zpad32(value):
+    return just(value, 32, b'\x00')
+
+
+def zpad_right(value, length):
+    return just(value, length, b'\x00', direction='right')
+
+
+def zpad32_right(value):
+    return just(value, 32, b'\x00', direction='right')
```
Comment (carver) [inline eth_abi/utils/padding.py]: The `just` function is not improving readability here, for me. I think all of these are best replaced with the `ljust` or `rjust` builtin calls.

For example, here that would be: `return value.ljust(32, b'\x00')`.

File: eth_abi/utils/padding.py [hunk 2]
```diff
@@ -1,15 +1,33 @@
-import functools
+def just(value, length, fillerchar, direction='left'):
+    if isinstance(value, str):
+        _fillerchar = fillerchar.decode()
```
Comment (carver) [inline eth_abi/utils/padding.py]: I think it's better to just fix the calls that are using a fillerchar of the wrong type.

**Comment by carver** [comment]
I think the best way to add the eth-hash backend (to fix the broken test) is like: https://github.com/ethereum/eth-keys/pull/35/commits/fd72ba1e9875c39b33f25fc0b5a15df6b362e508

_Well, actually, I think the best way is to set up a `tester` extra and get rid of requirements-dev altogether, but that seems out of scope for this PR._

**Review by carver — APPROVED** [review]

**Comment by pipermerriam** [comment]
I'm taking this over.  There were some deeper nasty things that required fixing in conjunction. almost done.

**Comment by carver** [comment]
Yeah, solving those issues was *much* better than hiding the encoding in the zpadding. Latest changes look good :+1:

I'm happy to look again, if any more major changes are needed after the test results come in.

**Comment by dylanjw** [comment]
@pipermerriam

Installing this branch in my web3.py environment has tests failing on another missing import from eth_util.

`coerce_args_to_text` import in `eth_abi/utils/parsing.py`.  It looks like `coerce_args_to_text` was removed from the latest eth_utils.

**Comment by dylanjw** [comment]
Maybe there is something wrong with my environment.  I justs ran the eth-abi tests, saw the missing import error, and the re-ran in a new virtual environment and got different test results

**Comment by dylanjw** [comment]
Ok, so I was installing the latest eth-utils somehow, which has `coerce_args_to_text` removed from eth_utils/strings.py.

With a clean environment, I am getting a different result than the travis tests:

```
====================================== FAILURES =======================================
_______________________________ test_decode_signed_real _______________________________

    @settings(max_examples=1000)
>   @given(
        high_bit_size=st.integers(min_value=1, max_value=32).map(lambda v: v * 8),
        low_bit_size=st.integers(min_value=1, max_value=32).map(lambda v: v * 8),
        integer_bit_size=st.integers(min_value=1, max_value=32).map(lambda v: v * 8),
        stream_bytes=st.binary(min_size=0, max_size=32, average_size=32),
        data_byte_size=st.integers(min_value=0, max_value=32),
    )

f          = <function given.<locals>.run_test_with_generator.<locals>.wrapped_test at 0x7fc786a57510>

tests/decoding/test_single_decoders.py:430: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../../.pyenv/versions/3.5.4/lib/python3.5/site-packages/hypothesis/core.py:578: in execute
    result = self.test_runner(data, run)
../../../../.pyenv/versions/3.5.4/lib/python3.5/site-packages/hypothesis/executors.py:58: in default_new_style_executor
    return function(data)
../../../../.pyenv/versions/3.5.4/lib/python3.5/site-packages/hypothesis/core.py:570: in run
    return test(*args, **kwargs)
tests/decoding/test_single_decoders.py:430: in test_decode_signed_real
    @given(
../../../../.pyenv/versions/3.5.4/lib/python3.5/site-packages/hypothesis/core.py:517: in test
    result = self.test(*args, **kwargs)
tests/decoding/test_single_decoders.py:480: in test_decode_signed_real
    decoded_value = decoder(stream)
eth_abi/decoding.py:102: in __call__
    return self.decode(stream)
eth_abi/decoding.py:168: in decode
    cls.validate_padding_bytes(value, padding_bytes)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cls = <class 'eth_abi.decoding.SignedRealDecoder'>, value = Decimal('-128.000')
padding_bytes = b'\x00'

    @classmethod
    def validate_padding_bytes(cls, value, padding_bytes):
        value_byte_size = cls._get_value_byte_size()
        padding_size = cls.data_byte_size - value_byte_size
    
        if value >= 0:
            expected_padding_bytes = b'\x00' * padding_size
        else:
            expected_padding_bytes = b'\xff' * padding_size
    
        if padding_bytes != expected_padding_bytes:
            raise NonEmptyPaddingBytes(
>               "Padding bytes were not empty: {0}".format(repr(padding_bytes))
            )
E           eth_abi.exceptions.NonEmptyPaddingBytes: Padding bytes were not empty: b'\x00'

cls        = <class 'eth_abi.decoding.SignedRealDecoder'>
expected_padding_bytes = b'\xff'
padding_bytes = b'\x00'
padding_size = 1
value      = Decimal('-128.000')
value_byte_size = 2

eth_abi/decoding.py:421: NonEmptyPaddingBytes
------------------------------------- Hypothesis --------------------------------------
Falsifying example: test_decode_signed_real(high_bit_size=8, low_bit_size=8, integer_bit_size=16, stream_bytes=b'\x00\x80\x00', data_byte_size=3)
======================= 1 failed, 115 passed in 126.91 seconds ========================
```

**Pull request merged**